custom/icds/views.py

```diff
@@ -1,31 +1,50 @@
+import requests
+
 from django.http import HttpResponse
 from django.template.loader import render_to_string
 from corehq import toggles
 from corehq.apps.domain.decorators import login_and_domain_required
 from corehq.apps.locations.models import SQLLocation
+from . import settings
+from .exceptions import TableauTokenException
 
 
 @toggles.ICDS_REPORTS.required_decorator()
 @login_and_domain_required
 def tableau(request, domain, workbook, worksheet):
     # TODO: In production we should limit this to only the actual workbook, but this makes iteration much easier
-    # trusted_ticket = _get_tableau_trusted_ticket(request.)
-    couch_user = getattr(request, 'couch_user', None)
-    location_type_name, location_name = _get_user_permissions(couch_user, domain)
-
     context = {
         'report_workbook': workbook,
         'report_worksheet': worksheet,
-        'view_by': location_type_name,
-        'view_by_value': location_name,
         'debug': request.GET.get('debug', False),
     }
 
+    # set report view-by level based on user's location level
+    couch_user = getattr(request, 'couch_user', None)
+    location_type_name, location_name = _get_user_location(couch_user, domain)
+    context.update({
+        'view_by': location_type_name,
+        'view_by_value': location_name,
+    })
+
+    # set Tableau access URL
+    if request.GET.get('with_login', False):
+        # TABLEAU_VIEW_URL requires user to login before accessing views
+        tableau_access_url = settings.TABLEAU_VIEW_URL
+    else:
+        # generate a login free access URL
+        # This doesn't work on local as this header is added only by ngnix proxy
```

Comment:
Just FYI, you can install nginx and have it work as a proxy locally by following this [README](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/builds/README.md#install-nginx)